keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,277 @@
+        num_layers: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.
+        magnitude_standard_deviation: the standard deviation to use when drawing values
+            for the perturbations.
+        probability_to_apply:  the probability to apply an augmentation at each layer.
+
+    Usage:
+    ```python
+    (x_test, y_test), _ = tf.keras.datasets.cifar10.load_data()
+    rand_augment = keras_cv.layers.RandAugment(
+        value_range=(0, 255), num_layers=3, magnitude=5.0
+    )
+    x_test = rand_augment(x_test)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        num_layers=3,
+        magnitude=5.0,
+        magnitude_standard_deviation=1.5,
+        probability_to_apply=None,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.num_layers = num_layers
+        self.magnitude = magnitude
+        self.value_range = value_range
+        if magnitude < 0.0 or magnitude > 10.0:
+            raise ValueError(
+                f"`magnitude` must be in the range [0, 10], got `magnitude={magnitude}`"
+            )
+        self.magnitude_standard_deviation = magnitude_standard_deviation
+        self.probability_to_apply = probability_to_apply
+
+        policy = create_rand_augment_policy(magnitude, magnitude_standard_deviation)
+
+        self.auto_contrast = cv_preprocessing.AutoContrast(
+            **policy["auto_contrast"], value_range=(0, 255), seed=seed
+        )
+        self.equalize = cv_preprocessing.Equalization(
+            **policy["equalize"], value_range=(0, 255), seed=seed
+        )
+
+        self.solarize = cv_preprocessing.Solarization(
+            **policy["solarize"], value_range=(0, 255), seed=seed
+        )
+        self.solarize_add = cv_preprocessing.Solarization(
+            **policy["solarize_add"], value_range=(0, 255), seed=seed
+        )
+        self.invert = cv_preprocessing.Solarization(
+            **policy["invert"], value_range=(0, 255), seed=seed
+        )
+
+        self.color = cv_preprocessing.RandomColorDegeneration(
+            **policy["color"], seed=seed
+        )
+        self.contrast = cv_preprocessing.RandomContrast(**policy["contrast"], seed=seed)
+        self.brightness = cv_preprocessing.RandomBrightness(
+            **policy["brightness"], seed=seed
+        )
+        self.shear_x = cv_preprocessing.RandomShear(**policy["shear_x"], seed=seed)
+        self.shear_y = cv_preprocessing.RandomShear(**policy["shear_y"], seed=seed)
+        self.translate_x = cv_preprocessing.RandomTranslation(
+            **policy["translate_x"], seed=seed
+        )
+        self.translate_y = cv_preprocessing.RandomTranslation(
+            **policy["translate_y"], seed=seed
+        )
+        self.cutout = cv_preprocessing.RandomCutout(**policy["cutout"], seed=seed)
+
+        self.augmentation_layers = [
```

Comment:
We may want to factor this as a separate overridable property so as to make it configurable via easy subclassing.

Reply:
That is an API that may be nice.  I could see that being useful.  Maybe we should think about offering an AutoAugment class that takes a list of layers and produces a policy from it, then have RandAugment be a special case.